### PR TITLE
error_code: remove default implementation

### PR DESCRIPTION
**Pull request by gregwebs**
### What problem does this PR solve?

default <T> to unknown means the compiler cannot tell us if error codes
are working. It is easy to accidentally end up with the unknown code.
It also requires all implemntations to use specialization.

### What is changed and how it works?

Remove defaulting to unknown
use an error_unknown macro if there is no error code to set it to
unknown.

### Related changes

None

### Check List

Tests

Tested a case where the error code shows up after these changes. Previously wrappers were getting defaulted to unknown but are now properly dereferenced.

Side effects

Code may change from Unknown to a more specific code.

### Release note

Fix error code propagation

### CODE REVIEW

File: components/tikv_util/src/log.rs [hunk 1]
```diff
@@ -33,6 +33,25 @@ macro_rules! error {
     };
 }
 
+#[macro_export]
+macro_rules! error_unknown {
```
Comment (5kbpers) [inline components/tikv_util/src/log.rs:37]: Please add some comments to explain the intention of this macro

**Review by 5kbpers — COMMENTED** [review]
rest LGTM

File: components/tikv_util/src/log.rs (hunk 1 above)
Comment (gregwebs) [reply components/tikv_util/src/log.rs:41]: 👍 comment added.

**Review by 5kbpers — APPROVED** [review]

**Review by nrc — APPROVED** [review]
lgtm

**Pull request merged**